HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentComponentsController+Helpers.swift

```diff
@@ -0,0 +1,806 @@
+            presentedVC.dismiss(animated: true) { [weak self] in
+                guard let self = self, let navController = self.navigationControllerProvided else { return }
+                self.obtainPDFURLFromPaymentRequest(paymentInfo: paymentInfo, viewController: navController)
+            }
+        } else {
+            guard let presentedVC = self.navigationControllerProvided?.presentedViewController else { return }
+            self.obtainPDFURLFromPaymentRequest(paymentInfo: paymentInfo, viewController: presentedVC)
+        }
+    }
+    
+    private func processPaymentRequest(paymentInfo: GiniInternalPaymentSDK.PaymentInfo) {
+        createPaymentRequest(paymentInfo: paymentInfo) { [weak self] result in
+            switch result {
+            case .success(let paymentRequestID):
+                self?.handleSuccessfulPaymentRequest(paymentRequestID: paymentRequestID)
+            case .failure(let error):
+                self?.errors.append(error.localizedDescription)
+                self?.showErrorsIfAny()
+            }
+        }
+    }
+
+    private func handleSuccessfulPaymentRequest(paymentRequestID: String) {
+        if let presentedVC = navigationControllerProvided?.presentedViewController {
+            presentedVC.dismiss(animated: true) { [weak self] in
+                self?.openPaymentProviderApp(requestId: paymentRequestID)
+            }
+        } else {
+            openPaymentProviderApp(requestId: paymentRequestID)
+        }
+    }
+
+    private func openPaymentProviderApp(requestId: String) {
+        let universalLink = selectedPaymentProvider?.universalLinkIOS ?? ""
+        openPaymentProviderApp(requestId: requestId, universalLink: universalLink)
+    }
+
+    private func presentInstallAppBottomSheet() {
+        let installAppBottomSheet = installAppBottomSheet()
+        installAppBottomSheet.modalPresentationStyle = .overFullScreen
+        self.dismissAndPresent(viewController: installAppBottomSheet, animated: false)
+    }
+    
+    private func showErrorsIfAny() {
+        if !errors.isEmpty {
+            let uniqueErrorMessages = Array(Set(errors))
+            DispatchQueue.main.async {
+                self.delegate?.isLoadingStateChanged(isLoading: false)
+                self.showErrorAlertView(error: uniqueErrorMessages.joined(separator: ", "))
+            }
+            errors = []
+        }
+    }
+    
+    func showErrorAlertView(error: String) {
+        if navigationControllerProvided?.presentedViewController != nil {
+            self.navigationControllerProvided?.presentedViewController?.dismiss(animated: true, completion: {
+                self.presentAlerViewController(error: error)
+            })
+        } else {
+            presentAlerViewController(error: error)
+        }
+    }
+
+    private func presentAlerViewController(error: String) {
+        let alertController = UIAlertController(title: "Error test",
```

Comment:
[clarification] The error title should contain test or is it just an error here?

Reply:
No. It's fixed and replaced in next PR with general error message